tidy3d/components/geometry/base.py

```diff
@@ -2011,6 +2161,344 @@ def _surface_area(self, bounds: Bound) -> float:
+    ) -> List[Shapely]:
+        """Return a list of shapely geometries at the plane specified by normal and origin.
+
+        Parameters
+        ----------
+        normal : Coordinate
+            Vector defining the normal direction to the plane.
+        origin : Coordinate
+            Vector defining the plane origin.
+        to_2D : MatrixReal4x4
+            Transformation matrix to apply to resulting shapes.
+
+        Returns
+        -------
+        List[shapely.geometry.base.BaseGeometry]
+            List of 2D shapes that intersect plane.
+            For more details refer to
+            `Shapely's Documentaton <https://shapely.readthedocs.io/en/stable/project.html>`_.
+        """
+        return self.geometry.intersections_tilted_plane(
+            tuple(np.dot((normal[0], normal[1], normal[2], 0.0), self.transform)[:3]),
+            tuple(np.dot(self.inverse, (origin[0], origin[1], origin[2], 1.0))[:3]),
+            np.dot(to_2D, self.transform),
+        )
+
+    def inside(
+        self, x: np.ndarray[float], y: np.ndarray[float], z: np.ndarray[float]
+    ) -> np.ndarray[bool]:
+        """For input arrays ``x``, ``y``, ``z`` of arbitrary but identical shape, return an array
+        with the same shape which is ``True`` for every point in zip(x, y, z) that is inside the
+        volume of the :class:`Geometry`, and ``False`` otherwise.
+
+        Parameters
+        ----------
+        x : np.ndarray[float]
+            Array of point positions in x direction.
+        y : np.ndarray[float]
+            Array of point positions in y direction.
+        z : np.ndarray[float]
+            Array of point positions in z direction.
+
+        Returns
+        -------
+        np.ndarray[bool]
+            ``True`` for every point that is inside the geometry.
+        """
+        x = np.array(x)
+        y = np.array(y)
+        z = np.array(z)
+        xyz = np.dot(self.inverse, np.vstack((x.flat, y.flat, z.flat, np.ones(x.size))))
+        if xyz.shape[1] == 1:
+            # TODO: This "fix" is required because of a bug in PolySlab.inside (with non-zero sidewall angle)
+            return self.geometry.inside(xyz[0][0], xyz[1][0], xyz[2][0]).reshape(x.shape)
+        return self.geometry.inside(xyz[0], xyz[1], xyz[2]).reshape(x.shape)
+
+    def _volume(self, bounds: Bound) -> float:
+        """Returns object's volume within given bounds."""
+        # NOTE (Lucas): Bounds are overestimated.
+        vertices = np.dot(self.inverse, self._vertices_from_bounds(bounds))[:3]
+        inverse_bounds = (tuple(vertices.min(axis=1)), tuple(vertices.max(axis=1)))
+        return abs(np.linalg.det(self.transform)) * self.geometry.volume(inverse_bounds)
+
+    def _surface_area(self, bounds: Bound) -> float:
+        """Returns object's surface area within given bounds."""
+        log.warning("Surface area of transformed elements cannot be calculated.")
+        return 0.0
```

Comment:
Would it make sense to `return None` here? in case someone misses the warning, this will throw a wrench in their operations using the surface area and make it more explicit. I see value of returning 0.0 too but could you explain the reasoning?

Reply:
Zero was just arbitrary. I agree with returning `None` to make it even more explicit.